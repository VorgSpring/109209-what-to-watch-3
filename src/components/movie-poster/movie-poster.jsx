import React from 'react';
import PropTypes from 'prop-types';

export const MoviePoster = ({poster, name, isBig, children}) => (
  <div className='movie-card__info'>
    <div className={`movie-card__poster ${isBig && `movie-card__poster--big`}`}>
      <img
        src={poster}
        alt={`${name} poster`}
        width='218'
        height='327'
      />
    </div>
    {children}
  </div>
);

MoviePoster.defaultProps = {
  poster: ``,
  name: ``,
  isBig: false,
  children: []
};

MoviePoster.propTypes = {
  poster: PropTypes.string,
  name: PropTypes.string,
  isBig: PropTypes.bool,
  children: PropTypes.node
};
