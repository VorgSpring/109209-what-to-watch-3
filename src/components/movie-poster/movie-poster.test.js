import React from 'react';
import {MoviePoster} from './movie-poster.jsx';
import renderer from 'react-test-renderer';

describe(`MoviePoster`, () => {
  it(`should renders correctly`, () => {
    const tree = renderer.create(
        <MoviePoster />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
