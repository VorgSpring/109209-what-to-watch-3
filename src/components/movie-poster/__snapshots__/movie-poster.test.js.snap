// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePoster should renders correctly 1`] = `
<div
  className="movie-card__info"
>
  <div
    className="movie-card__poster false"
  >
    <img
      alt=" poster"
      height="327"
      src=""
      width="218"
    />
  </div>
</div>
`;
